virtcontainers/ipvlan_endpoint.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"fmt"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+)
+
+// IPVlanEndpoint represents a macvlan endpoint that is bridged to the VM
+type IPVlanEndpoint struct {
+	NetPair            NetworkInterfacePair
+	EndpointProperties NetworkInfo
+	EndpointType       EndpointType
+	PCIAddr            string
+}
+
+func createIPVlanNetworkEndpoint(idx int, ifName string) (*IPVlanEndpoint, error) {
+	if idx < 0 {
+		return &IPVlanEndpoint{}, fmt.Errorf("invalid network endpoint index: %d", idx)
+	}
+
+	// Use tc filtering for ipvlan, since the other inter networking models will
```

Comment:
@amshinde are there any other cases you know of where macvlan or bridge does not work. Ideally you need to have a check that sees if the current interworkingModel works for this type of interface. This function can return what interworkingModel will support it, if the user selected one does not. This will make the code generic.

Reply:
@mcastelino Yes, I need to handle this for macvlan as well. I agree this code needs to be generic.
I think it will make sense to have an interface for each endpoint that advertises the internetworking models it supports. I will add that for every endpoint so that we can make a decision to fallback to a model that works for an endpoint. 
I'll do this in a separate PR.